dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaYarnApplicationsTask.java

```diff
@@ -0,0 +1,177 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.ByteSink;
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.ClouderaManagerHandle.ClouderaClusterDTO;
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.dto.ApiYARNApplicationDTO;
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.dto.ApiYARNApplicationListDTO;
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.dto.ApiYARNApplicationTypeDTO;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.io.IOException;
+import java.io.Writer;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** The task dumps YARN applications from Cloudera Manager API */
+public class ClouderaYarnApplicationsTask extends AbstractClouderaManagerTask {
+  private static final Logger LOG = LoggerFactory.getLogger(ClouderaYarnApplicationsTask.class);
+  private static final DateTimeFormatter isoDateTimeFormatter =
+      DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+
+  private final ImmutableList<String> predefinedAppTypes = ImmutableList.of("MAPREDUCE", "SPARK");
+  private final int includedLastDays;
+
+  public ClouderaYarnApplicationsTask(int days) {
+    super(String.format("yarn-applications-%d.jsonl", days));
+    this.includedLastDays = days;
+  }
+
+  @Override
+  protected void doRun(
+      TaskRunContext context, @Nonnull ByteSink sink, @Nonnull ClouderaManagerHandle handle)
+      throws Exception {
+    List<ClouderaClusterDTO> clusters = handle.getClusters();
+    Preconditions.checkNotNull(
+        clusters, "Clusters must be initialized before fetching YARN applications.");
+
+    List<ApiYARNApplicationDTO> totalYARNApplications = new ArrayList<>();
+    YarnApplicationsLoader appLoader =
+        new YarnApplicationsLoader(handle.getApiURI().toString(), handle.getHttpClient());
+    for (ClouderaClusterDTO cluster : handle.getClusters()) {
```

Comment:
You already have `List<ClouderaClusterDTO> clusters` variable above. Please reuse, `handle.getClusters()` is synchronised.